public/app/features/teams/state/actions.ts

```diff
@@ -223,6 +223,7 @@ export function updateTeamPreferences() {
     const preferences = getStore().team.preferences;
 
     await getBackendSrv().put(`/api/teams/${team.id}/preferences`, preferences);
-    window.location.reload();
+
+    dispatch(loadTeamPreferences());
```

Comment:
Not sure this is the solution. The idea is to do a full page reload in case the theme are changed. That's how it works for org and user preferences updates.